### PR TITLE
Add annotation setup

### LINKED ISSUE #1980: Class attribute type annotations
## Feature

The ability to annotate a type on a class attribute. For example:

```python
class Employee:
    name: str
    id: int
```

This runs without error with CPython, but raises the following error with RustPython:

```
Traceback (most recent call last):
  File "test.py", line 3, in <module>
    id: int
  File "test.py", line 2, in Employee
    name: str
NameError: name '__annotations__' is not defined
```

## Python Documentation

https://docs.python.org/3/library/typing.html#typing.NamedTuple

### PR DESCRIPTION
**Pull request by Lynskylate**
Fix #1980 
Add setup annotation for compile class

### CODE REVIEW

File: compiler/src/compile.rs [hunk 1]
```diff
@@ -984,6 +984,64 @@ impl<O: OutputStream> Compiler<O> {
...
     }
 
+    fn option_stmt_to_bool(&mut self, stmts: &Option<ast::Suite>) -> bool {
+        match &stmts {
+            Some(stmts) => self.find_ann(stmts),
+            None => false,
+        }
+    }
+
+    fn find_ann(&mut self, body: &[ast::Statement]) -> bool {
```
Comment (coolreader18) [inline compiler/src/compile.rs:994]: I don't think this needs to be `&mut self`:
```suggestion
    fn find_ann(&self, body: &[ast::Statement]) -> bool {
```

File: compiler/src/compile.rs [hunk 2]
```diff
@@ -984,6 +984,64 @@ impl<O: OutputStream> Compiler<O> {
         Ok(())
     }
 
+    fn option_stmt_to_bool(&mut self, stmts: &Option<ast::Suite>) -> bool {
+        match &stmts {
+            Some(stmts) => self.find_ann(stmts),
+            None => false,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Lynskylate) [reply compiler/src/compile.rs]: Done, thank you.
Before, I mistakenly made self mut, which caused can't use closure function.

File: compiler/src/compile.rs (hunk 1 above)
Comment (Lynskylate) [reply compiler/src/compile.rs]: Done, Thank you

File: vm/src/frame.rs [hunk 3]
```diff
@@ -456,6 +456,13 @@ impl ExecutingFrame<'_> {
                 Ok(Some(ExecutionResult::Yield(value)))
             }
             bytecode::Instruction::YieldFrom => self.execute_yield_from(vm),
+            bytecode::Instruction::SetupAnnotation => {
+                let locals = self.scope.get_locals();
+                if let Err(_e) = locals.get_item("__annotations__", vm) {
```
Comment (coolreader18) [inline vm/src/frame.rs:461]:
```suggestion
                if !locals.contains_key("__annotations__") {
```

**Comment by coolreader18** [comment]
Thanks for contributing!

**Pull request merged**